File: scala/9.scala
var result = 0;

for (
    a <- 1 to 332;
    b <- a to ((1000 - a) / 2) - 1;
    c =  1000 - a - b
    if (a * a + b * b == c * c))
  result = a * b *c

println(result)
assert(result == 31875000)

File: haskell/9.hs
triple = head [[a,b,c] | a <- [1..332],
                         b <- [a..((1000-a) `div` 2)-1],
                         c <- [1000-a-b],
                         a+b+c == 1000,
                         a^2+b^2 == c^2]
result = product triple
main = print result
